sql-plugin/src/main/scala/org/apache/spark/sql/rapids/stringFunctions.scala

```diff
@@ -1341,6 +1341,22 @@ abstract class StringSplitRegExpMeta[INPUT <: TernaryExpression](expr: INPUT,
       } else {
         if (utf8Str.numChars() == 0) {
           willNotWorkOnGpu("An empty delimiter pattern is not supported")
+        }
+        isRegExp = RegexParser.isRegExpString(utf8Str.toString)
+        if (isRegExp) {
+          val transpiler = new CudfRegexTranspiler(RegexSplitMode)
+          transpiler.transpileToSplittableString(utf8Str.toString) match {
+            case Some(simplified) =>
+              pattern = simplified
+              isRegExp = false
+            case None =>
+              try {
+                pattern = transpiler.transpile(utf8Str.toString)
+              } catch {
+                case e: RegexUnsupportedException =>
+                  willNotWorkOnGpu(e.getMessage)
+              }
+          }
         } else {
           val str = utf8Str.toString
           isRegExp = RegexParser.isRegExpString(str)
```

Comment:
This line appears to be redundant. `isRegExp` is already initialized prior to the `if` statement so it is guaranteed to be `false` if we are hitting the `else` block?

``` scala
        isRegExp = RegexParser.isRegExpString(utf8Str.toString)
        if (isRegExp) {
```

Reply:
In this case, I did go by code that was already in place, also `isRegExp` is declared as a variable (`var`):

```
 var isRegExp: Boolean = false
```

I could rewrite this code to be a little cleaner though

Reply:
never mind, I see the redundancy. I missed this when merging in upstream code.

Reply:
Cleaned this up in latest commit